modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java

```diff
@@ -407,32 +407,51 @@ else if ((nestedFieldsDepth > 0) &&
 
 			objectRelationships.forEach(
 				objectRelationship -> {
-					Object[] objectEntries = new ObjectEntry[0];
+					Optional<UriInfo> uriInfoOptional =
+						dtoConverterContext.getUriInfoOptional();
 
-					if (Objects.equals(
-							objectRelationship.getType(),
-							ObjectRelationshipConstants.TYPE_MANY_TO_MANY)) {
+					if (uriInfoOptional.map(
+							UriInfo::getQueryParameters
+						).map(
+							queryParameters -> queryParameters.getFirst(
+								"nestedFields")
+						).map(
+							nestedFields -> nestedFields.contains(
```

Comment:
I think this contains could be mistaken. For example:

relationshipName1 -> `alumnoTutor`
relationshipName2 -> `alumnoTutores`
nestedFields -> `alumnoTutores`

`nestedFields.contains("alumnoTutor") `-> true

Reply:
Yes, good catch, I will fix this. Thanks

Reply:
I think there is another appearance in the same method so maybe we could fix it there as well

Reply:
Yes, there is another appearance but it is different in that case because the name is not the relationship name. In fact, is a behavior that we want to change but we agreed to do it in another PR to separate the logic between PRs.